### ISSUE TITLE
conceptual question about the embeddings

### ISSUE THREAD

**Issue by randomgambit** [UKPLab/sentence-transformers#1326]
Hello there! and thanks for this package. It is really super fast and efficient.

I just have a conceptual question about the models that are available in `sentence-transformers`. Are they trained for a particular task or they come from big unsupervised masked language models (say BERT) trained on massive amounts of text?

For instance, if I were to train a BERT masked language model from scratch using `huggingface` I would also obtain embeddings at the sentence level. Would these embeddings be similar to the ones available in `sentence-transformers`? The reason I am asking is because I want to trained a masked language model on my highly specialized corpus of text.

Thanks!

**Comment by nreimers** [comment]
See the Sentence-BERT paper.

Just training MLM does not work well - you will get a bad performance. Supervision on labeled data is needed

**Comment by hyusterr** [comment]
Following this question, I'm wondering when training SBERT we are actually updating parameters of one BERT or 2 BERT?
if two, then which one is the one we use to get sentence embeddings?

**Comment by nreimers** [comment]
Only one

**Comment by hyusterr** [comment]
thanks for your instant reply!
so when backpropagating loss, the gradient pass through the whole transformer 2 times? for u and v  (follow the papers notation)

**Comment by nreimers** [comment]
Yes

**Comment by hyusterr** [comment]
thanks for solving my confusion so fast :)
